### DISCUSSION TITLE
How does Shouldly attach methods to everything

### DISCUSSION THREAD

**Discussion by gwasserfall** [shouldly/shouldly#832] [Q&A] [answered]
Hello Shouldly community, how are you? :)

I was wondering how Shouldy can attached the Should(x) method to all classes. I had a quick gander at the source and tests but couldn't find a definitive answer. I've been working mostly in Python and I would imagine that you would do something like sub classing the base Object class to attach methods as everything would inherit from this class.

If you could point me to the right source file with a tldr (or just a tldr!) I would appreciate it!

Regards G

**Comment by gwasserfall** [comment]
By sub class I mean override

**Comment by slang25** [comment, marked as answer]
Hi @gwasserfall, thanks for the question. In C# there is a handy feature called extension methods ([see here](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/extension-methods)), which lets us extend types with additional methods.

All we need to do is define a static method, which takes the type as the first parameter, and add the `this` modifier, like this:
```csharp
public static void ShouldBe(this string actual, string expected)
{
    // ...
}
```

When the type that contains this method is included by importing its containing namespace, the method appears just as if it were defined on the `string` type itself.

I hope I understood you correctly, let me know if not 🙂
